### ISSUE TITLE
KeyError when convert from Mark4 to VDIF format

### ISSUE THREAD

**Issue by shaoguangleo** [mhvk/baseband#330]
I am use the code test_conversation in this repo, and just replace the SAMPLE.m4 to a real data file.
The baseband version is 2.0.0.
When running, encounter the following error:

```
(bb) [leo@head1 src]$ python test_conversion.py rd1904_ft_128-1732.m5a
Traceback (most recent call last):
  File "test_conversion.py", line 440, in <module>
    test.test_stream('.', sys.argv[1])
  File "test_conversion.py", line 296, in test_stream
    data = fr.read(80000)  # full Mark 4 frame
  File "/home/leo/bb/lib/python2.7/site-packages/baseband/vlbi_base/base.py", line 566, in read
    data = frame[sample_offset:sample_offset + nsample]
  File "/home/leo/bb/lib/python2.7/site-packages/baseband/mark4/frame.py", line 261, in __getitem__
    data[ninvalid:] = self.payload[payload_item]
  File "/home/leo/bb/lib/python2.7/site-packages/baseband/vlbi_base/payload.py", line 253, in __getitem__
    decoder = self._decoders[self._coder]
KeyError: (16, 2, 2)
```

Looks like the frame length was not fit?

**Comment by mhvk** [comment]
@shaoguangleo - thanks for reporting! The error means there is not yet a decoder for `nchan=16`, `bps=2`, `fanout=2` (the error message could do with improving...). We've been adding these as we encounter data that actually has a given format, since then we can check with `mark5access` that we do it right. Before anything else, do your data indeed have 16 channels and bps=2?

In principle, it is not difficult to add support: one can just copy & adjust one of the other decoders (e.g., https://github.com/mhvk/baseband/blob/master/baseband/mark4/payload.py#L159). If you can do that yourself and make a PR, that would be great. Alternatively, if you could extract, say, the first 256kB of your file and make it accessible, I can see if I can quickly do it for you (I'm leaving on holidays coming sunday, so it may not be possible).

**Comment by shaoguangleo** [comment]
@mhvk Thanks for your response. I'm also on vacation last week, sorry for this.Just come to work today.
The data format (nchan=16, bps=2, fanout=2) is from Fortleza(Ft) station from Brazil.
There are several sessions of this data, so I think it is necessary to add this support.
As you mentioned, I will first check the mark4 payload, and also I will split the first 256kB data to upload.

**Comment by mhvk** [comment]
fixed by #380
